rdf-delta-server-local/src/main/java/org/seaborne/delta/server/local/patchstores/zk/PatchStoreProviderZk.java

```diff
@@ -17,33 +17,47 @@
 
 package org.seaborne.delta.server.local.patchstores.zk;
 
-import org.apache.curator.framework.CuratorFramework;
 import org.apache.jena.atlas.logging.Log;
+import org.apache.zookeeper.KeeperException;
 import org.seaborne.delta.DeltaConst;
 import org.seaborne.delta.server.Provider;
 import org.seaborne.delta.server.local.DPS;
 import org.seaborne.delta.server.local.LocalServerConfig;
 import org.seaborne.delta.server.local.PatchStore;
 import org.seaborne.delta.server.local.PatchStoreProvider;
-import org.seaborne.delta.zk.Zk;
+import org.seaborne.delta.zk.UncheckedZkConnection;
+import org.seaborne.delta.zk.ZkException;
+import org.seaborne.delta.zk.WrappedUncheckedZkConnection;
+import org.seaborne.delta.zk.direct.DirectZkConnection;
+
+import java.io.IOException;
 
 public class PatchStoreProviderZk implements PatchStoreProvider {
 
     public PatchStoreProviderZk() { }
 
     @Override
     public PatchStore create(LocalServerConfig config) {
-        CuratorFramework client = curator(config);
+        UncheckedZkConnection client = zk(config);
         return new PatchStoreZk(client, this);
     }
 
-    /** Build a {@link CuratorFramework} from the {@link LocalServerConfig}. */
-    protected CuratorFramework curator(LocalServerConfig config) {
+    /**
+     * Build an {@link UncheckedZkConnection} from the {@link LocalServerConfig}.
+     * @return A new {@link UncheckedZkConnection}.
+     * */
+    protected UncheckedZkConnection zk(LocalServerConfig config) {
         String connectionString = config.getProperty(DeltaConst.pDeltaZk);
         if ( connectionString == null )
             Log.error(this, "No connection string in configuration");
-        CuratorFramework client = Zk.curator(connectionString);
-        return client;
+        try {
+            return new WrappedUncheckedZkConnection(DirectZkConnection.connect(connectionString));
+        } catch (final IOException | KeeperException | InterruptedException e) {
```

Comment:
All the checked exceptions are caught, and any unchecked exceptions will already have the desired effect which is to halt the application. That's the only reason I'm wrapping the checked exceptions with a `ZkException`, which is unchecked. We need it to halt when it can't connect so the health checks fail, triggering its removal and replacement in the auto-scaling group. Before this change, it would enter a state where it would respond as healthy to pings and hang on request attempts which would result in support tickets in production. That was thanks to the Curator team deciding it was prudent to swallow connection exceptions entirely and just keep retrying as if the configuration is always guaranteed to be valid. This is why exception handling should always be deferred to the highest scope that would be interested in the event, but that's a rant for another time.

Reply:
For reasons unknown to me, this particular class relies on Jena logging when the others do not, so I think I skipped it to avoid addressing that logging inconsistency, but I've added it in the latest push.